gremlin-javascript/src/main/javascript/gremlin-javascript/lib/driver/connection.js

```diff
@@ -163,7 +163,13 @@ class Connection extends EventEmitter {
 
     this._ws.addEventListener('open', this.#handleOpen);
     this._ws.addEventListener('error', this.#handleError);
-    this._ws.on('unexpected-response', this.#handleUnexpectedResponse);
+    // Only attach unexpected-response listener for Node.js WebSocket (ws package)
+    // Browser WebSocket does not have this event and .on() method
+    if (!useGlobalWebSocket) {
```

Comment:
Can the check be changed to be consistent with the`off` condition?
```suggestion
    if (this._ws && 'on' in this._ws) {
```